### DISCUSSION TITLE
v1.1-rc1

### DISCUSSION THREAD

**Discussion by Chaarangan** [Chaarangan/NERP#28] [General]
Welcome to the March 2023 release of NERP. There are many updates in this version that we hope you'll like, some of the key highlights include

- Directory view - rearrange the codebase to make it more readable and understandable.
- Input files - support for testing on multiple files.
- Training usability - add variables to switch training mode, validation batch size, and shuffle training sets.
- Reproducibility improvements - fixed seed for data preparation, and model initiation.
- Performance Measures - Improved version of loss computation algorithms.

### Install 
The release will be available on:
- via pip
```bash
pip install NERP==1.1-rc1
```

-  or via the repository:
```bash
git clone --branch v1.1-rc1 https://github.com/Chaarangan/NERP.git
cd NERP && pip install -e .
```

### Training
- Common function to call
```python
model.train()
```
There are several options depending on your needs:
- Casual Training: Configure the YAML file and set ```continue_from_checkpoint``` as ```False``` and ```is_kfold``` as ```False```. Then call ```model.train()```.
- Training from a previous checkpoint: Configure the YAML file and set ```continue_from_checkpoint``` as ```True``` and ```is_kfold``` as ```False```. You will need to specify the ```checkpoint_path```. Then call ```model.train()```.
- Training with KFold: Configure the YAML file and set ```continue_from_checkpoint``` as ```False``` and ```is_kfold``` as ```True```. You will need to specify the number of ```splits```. If you wish to test each fold with your original test set rather than its own test split, set the ```test_on_original``` variable as ```True```. Then call ```model.train()```.
- Training from a previous checkpoint with KFold: Configure the YAML file and set ```continue_from_checkpoint``` as ```True``` and ```is_kfold``` as ```True```. You will need to specify the ```checkpoint_path```. Then call ```model.train()```.

### Predictions 
There are several options depending on your needs:
- Prediction on a CSV file: Configure the YAML file and give ```model_path```, ```tokenizer_path``` (if exists), ```in_file_path```, and ```out_file_path```. Then call ```model.predict()```.
```python
model.predict()
```

- Prediction on text: Configure the YAML file and give ```model_path``` and ```tokenizer_path``` (if exists). Then call ```model.predict_text(“some text”)```.
```python
output = model.predict_text("Hello from NERP")
print(output)
```

### Thank you
Last but certainly not least, a big Thank You to the contributors of NERP.

#### PRs
- [@tanmaysurana](https://github.com/tanmaysurana) (Tanmay Surana): add support for testing on multiple files, add additional parameters to maintain consistency across multiple experiments (validation batch size, shuffle, fixed seed), and improve loss computation algorithms [PR #20](https://github.com/Chaarangan/NERP/pull/20)

<hr /><em>This discussion was created from the release <a href='https://github.com/Chaarangan/NERP/releases/tag/v1.1-rc1'>v1.1-rc1</a>.</em>
